### PR TITLE
Fix a few problems with notifications on OstreeAsyncProgress

### PR DESCRIPTION
**Pull request by pwithnall**
As spotted by @dbnicholson in PR #819.

### CODE REVIEW

File: src/libostree/ostree-async-progress.c [hunk 1]
```diff
@@ -331,6 +331,8 @@ ostree_async_progress_set (OstreeAsyncProgress *self,
   if (self->dead)
     goto out;
 
+  changed = FALSE;
```
Comment (cgwalters) [inline src/libostree/ostree-async-progress.c]: Why not init this at variable decl time instead?

Comment (pwithnall) [inline src/libostree/ostree-async-progress.c]: I personally associate the assignment here more closely with the block below than with the actual function. Put another way, if there was a second block after this one which also uses `changed` (separately), then I wouldn’t want to rely on initialisation at declaration time for it.

But I’m not really fussed either way. I’ll change it if you want.

**Comment by cgwalters** [comment]
OK.

@rh-atomic-bot r+ 0d521ca

**Comment by cgwalters** [comment]
@rh-atomic-bot r+ 0d521ca

**Comment by dbnicholson** [comment]
This fixed the couple issues I was having. Thanks!

**Pull request closed without merging**